src/create-jobs.js

```diff
@@ -157,6 +157,48 @@ function checks (config) {
                 return createQaJobs('./vendor/bin/phpbench run --revs=2 --iterations=2 --report=aggregate', config);
             }
         ),
+        new Check(
+            config.code_checks,
+            [
+                function () {
+                    const composerFile = JSON.parse(fs.readFileSync('composer.json'));
+
+                    return (
+                        fs.existsSync('infection.json') ||
+                        fs.existsSync('infection.json.dist')
+                    ) && ! (
+                        // Skip executing infection, if roave/infection-static-analysis-plugin (redundant task)
+                        composerFile.hasOwnProperty('require-dev') &&
+                        composerFile['require-dev'].hasOwnProperty('roave/infection-static-analysis-plugin')
+                    );
+                }
+            ],
+            /**
+             * @param {Config} config
+             * @return {Array}
+             */
+            function (config) {
+                return createQaJobs('./vendor/bin/infection', config);
+            }
+        ),
+        new Check(
+            config.code_checks,
+            [
+                function () {
+                    const composerFile = JSON.parse(fs.readFileSync('composer.json'));
+
+                    return composerFile.hasOwnProperty('require-dev') &&
+                        composerFile['require-dev'].hasOwnProperty('roave/infection-static-analysis-plugin');
```

Comment:
so does this plugin won't need the `infection.json`?

Reply:
It does, but a better/simpler check is just to see if the dependency is there

Reply:
maybe checking both as for infection? you could check in infection and just swap the commands. So we end up having one new check rather than two which are mutually exclusive?